src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorDiagnosticsEndpoint.cs

```diff
@@ -179,6 +180,30 @@ public async Task<RazorDiagnosticsResponse> Handle(RazorDiagnosticsParams reques
             return filteredDiagnostics;
         }
 
+        private static bool InCSharpLiteral(OmniSharpVSDiagnostic d, SourceText sourceText, RazorSyntaxTree syntaxTree)
+        {
+            if (d.Range is null)
+            {
+                return false;
+            }
+
+            var owner = syntaxTree.GetOwner(sourceText, d.Range.End);
+            if (owner is null)
+            {
+                return false;
+            }
+
+            var isCSharp = owner.Kind switch
+            {
+                SyntaxKind.CSharpExpressionLiteral => true,
+                SyntaxKind.CSharpStatementLiteral => true,
+                SyntaxKind.CSharpEphemeralTextLiteral => true,
+                _ => false,
+            };
```

Comment:
Up to you, but this could be simply
```suggestion
            var isCSharp = owner.Kind is SyntaxKind.CSharpExpressionLiteral or SyntaxKind.CSharpStatementLiteral or SyntaxKind.CSharpEphemeralTextLiteral;
```

Reply:
oooo I didn't know about that syntax actually. Yes please!